stdlib/REPL/src/REPL.jl

```diff
@@ -177,15 +177,17 @@ function check_for_missing_packages_and_run_hooks(ast)
     end
 end
 
-function modules_to_be_loaded(ast, mods = Symbol[])
+function modules_to_be_loaded(ast::Expr, mods = Symbol[])
```

Comment:
```suggestion
modules_to_be_loaded(@nospecialize ast) = Symbol[]
function modules_to_be_loaded(ast::Expr, mods = Symbol[])
```
though perhaps it would better to check this in the other caller